homeassistant/components/graphite.py

```diff
@@ -0,0 +1,124 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Component that records all events and state changes and feeds the data to
+a graphite installation.
+
+Example configuration:
+
+ graphite:
+   host: foobar
+   port: 2003
+
+All options are optional, and assumed to be on localhost at the
+default port if not specified.
+"""
+import logging
+import threading
+import queue
+import socket
+import time
+
+from homeassistant.const import (
+    MATCH_ALL, EVENT_STATE_CHANGED,
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP,
+    STATE_ON, STATE_OFF)
+
+DOMAIN = "graphite"
+_LOGGER = logging.getLogger(__name__)
+_INSTANCE = None
+
+
+def setup(hass, config):
+    """ Setup graphite feeder. """
+    # pylint: disable=global-statement
+    global _INSTANCE
+
+    graphite_config = config.get('graphite', {})
+    host = graphite_config.get('host', 'localhost')
+    try:
+        port = int(graphite_config.get('port', 2003))
+    except ValueError:
+        _LOGGER.error('Invalid port specified')
+        return False
+
+    _INSTANCE = GraphiteFeeder(hass, host, port)
+    return True
+
+
+class GraphiteFeeder(threading.Thread):
+    """ Feeds data to graphite. """
+    def __init__(self, hass, host, port):
+        super(GraphiteFeeder, self).__init__(daemon=True)
+        self._hass = hass
+        self._host = host
+        self._port = port
+        self._queue = queue.Queue()
+        self._quit_object = object()
+
+        hass.bus.listen_once(EVENT_HOMEASSISTANT_START,
+                             self.start_listen)
+        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP,
+                             self.shutdown)
+        hass.bus.listen(MATCH_ALL, self.event_listener)
+
+    def start_listen(self, event):
+        """ Start event-processing thread. """
+        self.start()
+
+    def shutdown(self, event):
+        """ Tell the thread that we are done.
+
+        This does not block because there is nothing to
+        clean up (and no penalty for killing in-process
+        connections to graphite.
+        """
+        self._queue.put(self._quit_object)
+
+    def event_listener(self, event):
+        """ Queue an event for processing. """
+        self._queue.put(event)
+
+    def _send_to_graphite(self, data):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(10)
+        sock.connect((self._host, self._port))
+        sock.sendall(data.encode('ascii'))
+        sock.send('\n'.encode('ascii'))
+        sock.close()
+
+    def _report_attributes(self, entity_id, state, attributes):
+        now = time.time()
+        things = dict(attributes)
+        if isinstance(state, (float, int)):
+            state = float(state)
+        elif state in (STATE_ON, STATE_OFF):
+            state = float(state == STATE_ON)
+        else:
+            state = None
+        if state is not None:
+            things['state'] = state
+        lines = ['ha.%s.%s %f %i' % (entity_id, key.replace(' ', '_'),
```

Comment:
So as one of the graphite maintainers, I find that many users want their metrics with different prefixes. Instead of hardcoding the prefix as `ha`, is there any chance you could add this as a config option and default to `ha`?